mage/query-modules/cpp/map.md

```diff
@@ -0,0 +1,351 @@
+
+```plaintext
++----------------------------------------+
+| merged                                 |
++----------------------------------------+
+| {a: "b", c: "d", e: "f", g: "h"}       |
++----------------------------------------+
+```
+
+
+### `flatten(map, delimiter)`
+
+Flattens nested items in the input map.
+
+#### Input:
+
+- `map: Map[Any]` ➡ the map used in `flatten`
+- `delimiter: string (default = ".")` ➡ the delimiter used for flattening
+
+
+#### Output:
+
+- `result: Map[Any]` ➡ flattened map, sorted alphabetically by keys
+
+#### Usage:
+
+```cypher
+CALL map.flatten({a: {b:3, d:4}},"/") YIELD result RETURN result;
+```
+
+```plaintext
++----------------------------------------+
+| result                                 |
++----------------------------------------+
+| {"a/b": 3, "a/d": 4}                   |
++----------------------------------------+
+```
+
+### `from_lists(list_keys, list_values)`
+
+Makes a map from lists of keys and corresponding values.
+
+#### Input:
+
+- `list_keys: List[string]` ➡ the list of keys
+- `list_values` ➡ the list of values
+
+
+#### Output:
+
+- `result: Map[Any]` ➡ the resulting map
+
+#### Usage:
+
+```cypher
+CALL map.from_lists(["key","key2"],[1,2]) YIELD result RETURN result;
+```
+
+```plaintext
++----------------------------------------+
+| result                                 |
++----------------------------------------+
+| {""key": 1, "key2": 2}                 |
++----------------------------------------+
+```
+### `remove_keys(input_map, keys_list, recursive)`
+
+Removes keys from input map. If recursive option is true, will remove keys from maps nested inside the map. 
+
+#### Input:
+
+- `input_map: Map[Any]` ➡ the input map
+- `keys_list: List[string]` ➡ the list of keys which are to be removed
+- `recursive: boolean (default = false)` ➡ true if keys from nested map shall be removed, false if not
+
+
+#### Output:
+
+- `result: Map[Any]` ➡ the resulting map
+
+#### Usage:
+Usage with recursive = false:
+
+```cypher
+CALL map.remove_keys({key: 1, key2:{key : 3, key3: 5}},["key"],false) YIELD result RETURN result;
+```
+
+```plaintext
++----------------------------------------+
+| result                                 |
++----------------------------------------+
+|{"key2": {"key": 3,"key3": 5}}          |
++----------------------------------------+
+```
+
+
+Usage with recursive = true:
+
+```cypher
+CALL map.remove_keys({key: 1, key2:{key : 3, key3: 5}},["key"],true) YIELD result RETURN result;
+```
+
+```plaintext
++----------------------------------------+
+| result                                 |
++----------------------------------------+
+|{"key2": {"key3": 5}}                   |
++----------------------------------------+
+```
+
+### `from_nodes(label, property)`
+
+Returns a map of all nodes which contain the given label and property. The key of each map element will be the value of the property (if it is convertible to a string, otherwise throws ValueException).
```

Comment:
```suggestion
Returns a map of all nodes which contain the given label and property. The key of each map element will be the value of the property (if it is convertible to a string, otherwise throws `ValueException`).
```